src/compas/geometry/frame.py

```diff
@@ -24,6 +24,10 @@
 from .point import Point
 from .quaternion import Quaternion
 
+from scipy.spatial.transform import Rotation as R
+from scipy.spatial.transform import Slerp
```

Comment:
ah in discussing the issue #1208  you stated that:

> in general, if we can use something that already exists, we should. and since we stopped caring about IronPython, there is no need to make a pure Python version...

but I see the problem, and let's settle on a pure python version over conditional import of dependency ( eg switching between implementations depending on what python interpreter )

Reply:
you could also make the function a pluggable and add the `scipy` implementation as a plugin. then there is no conditionality and there is no need for the import being moved to the function/method body...

Reply:
only thing we can't do is break compatibility with IronPython in the core of COMPAS 2

Reply:
> you don't have to make the import conditional. just move it in to the function body

Shoot, I spend this morning reimplementing this in pure python 🐍
OK, it could be cool if that convention is made more explicit
Not sure if I like it: it creates an easy to miss discrepancy between what is supported across interpreters.

Reply:
Understood. I'm happy as things are: this is well tested and ready to be merged, if your content

Reply:
well with the imports at the top of the file, the entire `Frame` class is no longer available in IronPython. this should not be the case. so unfortunately, you will have to move the imports to the function body, or use a pur Python implementation, or use the a pluggable/plugin...

Reply:
thanks for the pluggable example!

Reply:
currently the workflows seem to fail because you still have the imports in `Frame.py` even though they are no longer being used...